p1, p2, p3, p4 Instance Properties of DOMQuad (#36700)
mdn/content: files/en-us/web/api/domquad/p1/index.md

---
title: "DOMQuad: p1 property"
short-title: p1
slug: Web/API/DOMQuad/p1
page-type: web-api-instance-property
browser-compat: api.DOMQuad.p1
---

{{APIRef("Geometry Interfaces")}}{{AvailableInWorkers}}

> The **`DOMQuad`** interface's **`p1`** property holds the {{domxref("DOMPoint")}} object that represents one of the four corners of the `DOMQuad`. This point includes the {{domxref("DOMPoint.x")}} (horizontal), {{domxref("DOMPoint.y")}} (vertical), {{domxref("DOMPoint.z")}} (depth), and {{domxref("DOMPoint.w")}} (perspective) coordinates for a point in space.

Reviewer: ```suggestion
The **`DOMQuad`** interface's **`p1`** property holds the {{domxref("DOMPoint")}} object that represents one of the four corners of the `DOMQuad`. When created from {{domxref("DOMQuad.fromRect()")}}, it is the point (x, y).
```

---
title: "DOMQuad: p1 property"
short-title: p1
slug: Web/API/DOMQuad/p1
page-type: web-api-instance-property
browser-compat: api.DOMQuad.p1
---

{{APIRef("Geometry Interfaces")}}{{AvailableInWorkers}}

The **`DOMQuad`** interface's

> **`p1`** property holds the {{domxref("DOMPoint")}} object that represents one of the four corners of the `DOMQuad`. This point includes the {{domxref("DOMPoint.x")}} (horizontal), {{domxref("DOMPoint.y")}} (vertical), {{domxref("DOMPoint.z")}} (depth), and {{domxref("DOMPoint.w")}} (perspective) coordinates for a point in space.

Reviewer: Could you remove these unnecessary line breaks? It's generally preferred to either break at each sentence or never break within paragraphs.

Also, is there any convention for _which_ corner each point is? For example, if you use `fromRect`, which point (bottom-left, etc.) is `p1`?

Author: - Will be removing line breaks from p1, p2, p3, and p4.
- To answer your question, there's no convention for which corner (https://drafts.fxtf.org/geometry/#dom-domquad-p1)
